libsolidity/interface/CompilerStack.cpp

```diff
@@ -150,10 +150,19 @@ bool CompilerStack::parse()
 			}
 		}
 	}
-	if (!Error::containsOnlyWarnings(m_errors))
-		// errors while parsing. should stop before type checking
+	if (Error::containsOnlyWarnings(m_errors))
+	{
+		m_stackState = ParsingSuccessful;
+		return true;
+	}
+	else
 		return false;
+}
 
+bool CompilerStack::analyze()
+{
+	if (m_stackState < SourcesSet)
```

Comment:
I think this requires `< ParsingSuccessful`. Also probably it makes sense to use equality here, e.g. allowing `analyze` only once.